tests/pipelines/test_pipelines_text_to_speech.py

```diff
@@ -0,0 +1,129 @@
+    import torch
+
+
+@is_pipeline_test
+@require_torch_or_tf
+class TextToSpeechPipelineTests(unittest.TestCase):
+    model_mapping = MODEL_FOR_TEXT_TO_SPEECH_MAPPING
+
+    @require_torch
+    def test_small_model_pt(self):
+        speech_generator = pipeline(
+            task="text-to-speech", model="microsoft/speecht5_tts", framework="pt", vocoder="microsoft/speecht5_hifigan"
+        )
+
+        # test if sampling_rate defined
+
+        # Using `do_sample=False` to force deterministic output
+        outputs = speech_generator("This is a test", do_sample=False)
+        self.assertEqual(
+            outputs,
+            ANY(torch.Tensor),
+        )
+
+        outputs = speech_generator(["This is a test", "This is a second test"])
+        self.assertEqual(
+            outputs,
+            [
+                ANY(torch.Tensor),
+                ANY(torch.Tensor),
+            ],
+        )
+
+        speaker_embeddings = torch.zeros((1, 512))
+
+        outputs = speech_generator(
+            "This is a test", do_sample=True, num_return_sequences=2, speaker_embeddings=speaker_embeddings
+        )
+        self.assertEqual(
+            outputs,
+            ANY(torch.Tensor),
+        )
+
+    @slow
+    @require_torch
+    def test_large_model_pt(self):
+        speech_generator = pipeline(task="text-to-speech", model="suno/bark-small", framework="pt")
+
+        # test if sampling_rate defined
+
+        # Using `do_sample=False` to force deterministic output
+        outputs = speech_generator("This is a test", do_sample=False, semantic_max_new_tokens=100)
+        self.assertEqual(
+            outputs,
+            ANY(torch.Tensor),
+        )
+
+        outputs = speech_generator(["This is a test", "This is a second test"])
+        self.assertEqual(
+            outputs,
+            [
+                ANY(torch.Tensor),
+                ANY(torch.Tensor),
+            ],
+        )
+
+        outputs = speech_generator(
+            "This is a test",
+            do_sample=True,
+            semantic_num_return_sequences=2,
+            speaker_embeddings="en_speaker_1",
+            semantic_max_new_tokens=100,
+        )
+        self.assertEqual(
+            outputs,
+            ANY(torch.Tensor),
+        )
+
+    def get_test_pipeline(self, model, tokenizer, processor):
+        speech_generator = TextToSpeechPipeline(model=model, tokenizer=tokenizer)
+        return speech_generator, ["This is a test", "Another test"]
+
+    def run_pipeline_test(self, speech_generator, _):
+        outputs = speech_generator("This is a test")
+        self.assertEqual(
+            outputs,
+            ANY(torch.Tensor),
+        )
+
+        outputs = speech_generator(["This is great !", "Something else"], num_return_sequences=2, do_sample=True)
+        self.assertEqual(
+            outputs,
+            [
+                ANY(torch.Tensor),
+                ANY(torch.Tensor),
+            ],
+        )
```

Comment:
This is not tested as long as there are no TTS models on https://huggingface.co/hf-internal-testing.

What should I do?

Reply:
I can run the script to create them. So it's the `Bark` model right?

Reply:
Hi @ydshieh , yes indeed! Many thanks!